akka-stream/src/main/scala/akka/stream/impl/io/TcpStages.scala

```diff
@@ -115,14 +115,31 @@ import scala.concurrent.{ Future, Promise }
         }
       }
 
-      setHandler(out, new OutHandler {
-        override def onPull(): Unit = {
-          // Ignore if still binding
-          if (listener ne null) listener ! ResumeAccepting(1)
-        }
+      setHandler(
+        out,
+        new OutHandler {
+          override def onPull(): Unit = {
+            // Ignore if still binding
+            if (listener ne null) listener ! ResumeAccepting(1)
+          }
 
-        override def onDownstreamFinish(): Unit = tryUnbind()
-      })
+          override def onDownstreamFinish(cause: Throwable): Unit = {
+            cause match {
+              case _: SubscriptionWithCancelException.NonFailureCancellation =>
+                log.debug(
+                  "Unbinding from {}:{} because downstream cancelled stream",
+                  endpoint.getHostString,
+                  endpoint.getPort)
+              case ex =>
+                log.error(
```

Comment:
Hmm, not sure if it should be an error, though? When error are handled correctly downstream that will most likely lead to double logging of the error. I guess I would be fine with debug logging In any case, I think it should not be more than a warning. Sorry, if it wasn't clear what I was suggesting before.

Reply:
Ok, I'll change to debug!